datastore/service.go

```diff
@@ -139,6 +111,61 @@ func (srv Service) DeleteResource(ctx context.Context, namespace models.Namespac
 	return repo.Delete(ctx, name)
 }
 
+func (srv Service) saveResource(
+	ctx context.Context,
+	namespace models.NamespaceSpec,
+	resourceSpecs []models.ResourceSpec,
+	obs progress.Observer,
+	execDatastore func(models.ResourceSpec) error,
+) error {
+	runner := parallel.NewRunner(parallel.WithLimit(ConcurrentLimit), parallel.WithTicket(ConcurrentTicketPerSec))
+	for _, incomingSpec := range resourceSpecs {
+		repo := srv.resourceRepoFactory.New(namespace, incomingSpec.Datastore)
+		runner.Add(func() (interface{}, error) {
+			proceed, err := srv.isProceedToSave(ctx, repo, incomingSpec)
+			if err != nil {
+				return nil, err
+			}
+
+			if !proceed {
+				srv.notifyProgress(obs, &EventResourceSkipped{
+					Spec:   incomingSpec,
+					Reason: "incoming resource is the same as existing",
+				})
+				return nil, nil // nolint:nilnil
+			}
+
+			if err := repo.Save(ctx, incomingSpec); err != nil {
+				return nil, err
+			}
+			return nil, execDatastore(incomingSpec)
+		})
+	}
+
+	var errorSet error
+	for _, result := range runner.Run() {
+		if result.Err != nil {
+			errorSet = multierror.Append(errorSet, result.Err)
+		}
+	}
+	return errorSet
+}
+
+func (Service) isProceedToSave(ctx context.Context, repo store.ResourceSpecRepository, incomingSpec models.ResourceSpec) (bool, error) {
```

Comment:
Can we rename the function from `isProceedToSave` to. `alreadyExists` and flip the boolean value from `proceed` to `exists`

Reply:
Or why not have equals method on resource struct and ignore id and urn fields with the same approach. Just use the equals method   by inlining the function.

Reply:
the intention is not to check whether the incoming exists or not. instead, it is to check whether the incoming is equal with the existing or not. I agree that using `isProceedToSave` should be changed. But, using `alreadyExists` also doesn't really reflect the intention. So, for now I just proposed to use `isEqualWithExisting`. Let me know if it works.

Reply:
updated based on the feedback by @sravankorumilli